### ISSUE TITLE
_text_similarity_metric_ops.so: undefined symbol

### ISSUE THREAD

**Issue by summa-code** [tensorflow/text#317]
Linux Ubuntu 20.04
TensorFlow installed from source latest nightly build for Cuda 11
TensorFlow-text version: Latest from source
Python version: 3.8.2
Installed using virtualenv? NO
Bazel version (if compiling from source): 3.1.0
GCC/Compiler version (if compiling from source): 9.3.0
CUDA/cuDNN version: 11.0
GPU model and memory: GeForce RTX 2070 8G

import tensorflow-text as tx
  File "<stdin>", line 1
    import tensorflow-text as tx
                     ^
SyntaxError: invalid syntax
>>> import tensorflow_text as tx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/xxxxx/.virtualenvs/tf-nightly/lib/python3.8/site-packages/tensorflow_text/__init__.py", line 21, in <module>
    from tensorflow_text.python import metrics
  File "/home/xxxxx/.virtualenvs/tf-nightly/lib/python3.8/site-packages/tensorflow_text/python/metrics/__init__.py", line 20, in <module>
    from tensorflow_text.python.metrics.text_similarity_metric_ops import *
  File "/home/xxxxx/.virtualenvs/tf-nightly/lib/python3.8/site-packages/tensorflow_text/python/metrics/text_similarity_metric_ops.py", line 28, in <module>
    gen_text_similarity_metric_ops = load_library.load_op_library(resource_loader.get_path_to_datafile('_text_similarity_metric_ops.so'))
  File "/home/xxxxx/.virtualenvs/tf-nightly/lib/python3.8/site-packages/tensorflow/python/framework/load_library.py", line 58, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename)
tensorflow.python.framework.errors_impl.NotFoundError: /home/xxxxx/.virtualenvs/tf-nightly/lib/python3.8/site-packages/tensorflow_text/python/metrics/_text_similarity_metric_ops.so: undefined symbol: _ZN10tensorflow14kernel_factory17OpKernelRegistrar12InitInternalEPKNS_9KernelDefEN4absl11string_viewESt10unique_ptrINS0_15OpKernelFactoryESt14default_deleteIS8_EE

**Comment by broken** [comment]
When you list the contents of that directory, do you see the shared object file?

```bash
ls /home/xxxxx/.virtualenvs/tf-nightly/lib/python3.8/site-packages/tensorflow_text/python/metrics
```

I would assume not. Did the wheel file you created when you built from source contain the *.so files? Can you list those files?

**Comment by broken** [comment]
Did you have your custom Tensorflow installed from source before trying to build TF Text? If not, the build will install TF 2.2.0 and build against that.
https://github.com/tensorflow/text/blob/master/oss_scripts/configure.sh#L29

If you did (which I suspect you did), I imagine the problem is that the op kernels are still building against tf 2.2.0 as defined in the workspace.
https://github.com/tensorflow/text/blob/master/WORKSPACE#L113

Can you update the WORKSPACE file to download whatever version of tf-nightly you used? I believe that should fix this.

**Comment by broken** [comment]
Make sure you run `bazel clean` before rebuilding tf text after you update the workspace file.

**Comment by broken** [comment]
We've updated to work with tf 2.3.x and released 2.3.0-rc1. This should be resolved.
